tools/jnimarshalmethod-gen/App.cs

```diff
@@ -56,43 +114,96 @@ static void CreateMarshalMethodAssembly (string path)
 			var destPath        = assemblyName.Name + ".dll";
 			var builder         = CreateExportedMemberBuilder ();
 
+			if (verbose)
+				Console.WriteLine ($"Preparing marshal method assembly '{assemblyName}'");
+
 			var da = AppDomain.CurrentDomain.DefineDynamicAssembly (
 					assemblyName,
 					AssemblyBuilderAccess.Save,
 					Path.GetDirectoryName (path));
 
 			var dm = da.DefineDynamicModule ("<default>", destPath);
 
+			var ad = resolver.GetAssembly (path);
```

Comment:
There is an "interesting" mix of Cecil & Reflection going on. Why? Why do we need to intermix both Cecil & Reflection here, instead of just using Reflection?

Reply:
I run into problems when using the Reflection and we discussed it in the meeting. Unfortunately I don't remember the details. So I opened an issue for that to look into it later https://github.com/xamarin/java.interop/issues/276